dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataOwnersManager.kt

```diff
@@ -211,13 +211,22 @@ class DataOwnersManager(
                 "User with id: ${userAuthentication.userId} is already a data owner of company with id: $companyId.",
             )
         }
-        emailSender.sendEmail(
-            dataOwnershipRequestEmailBuilder.buildDataOwnershipRequest(
-                companyId,
-                companyName,
-                userAuthentication,
-                comment,
-            ),
+        val properties = mapOf(
```

Comment:
Diese Funktion ist zu lang und macht zu viel. Ich würde folgende TEile in eigene sub-Funktionen stecken:

1) Den Teil der `companyRepository.findById()` macht und wenn nichts gefunden wird eine exception schmeißt
2) Den Teil der `dataOwnerRepository.findById()` macht und wenn es schon einen owner gibt eine exception schmeißt.

Dann würde ich auch das  "if necessary" aus dem Funktionennamen entfernen.

Reply:
sonar beschwert sich hier auch übrigens

Reply:
Ich hab die message Konstruktionslogik jetzt in die Methode vom message sender verschoben. Da war das eine findbyid zum getten des namens auch dabei. Bei dem habe ich das Errorhandling dann auch rausgeworfen.